### PR TITLE
Add io.github.ezQuake

### PR DESCRIPTION
**Pull request by Eonfge**
Related to #1160 

I had to choose between QuakeSpasm, Darkplaces, and ezQuake, and I choose for ezQuake as a first client to bring to Flatpak, mostly because its default installation is most feature complete. It uses SDL2 and Git for example, which is better when dealing with Flatpak.

This package does highlight something that I previously didn't account for, as most of these engines do not support user directories: The idea that there is an immutable game partition, and that changes are written into a /home/ directory, has escaped Quake and its source ports. As such, I've recreated this behaviour using some symlinking and shell scripting. Should also be compatible with nQuake-configuration since this is the same core engine.

Only thing I'm not sure of, is the screenshots. Upstream only provides 3:4 screenshots. Should I make a few new ones for this package?

### CODE REVIEW

**Comment by Eonfge** [comment]
bot, build io.github.ezQuake

**Comment by Eonfge** [comment]
bot, build io.github.ezQuake

File: single-click-setup.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
```
Comment (bilelmoussaoui) [inline single-click-setup.sh]: Please remove this file. Not needed on Flathub.

Comment (Eonfge) [reply single-click-setup.sh]: Removed it. I'll keep it local next time

**Comment by Eonfge** [comment]
Got one bug to look into

File: io.github.ezQuake.yaml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+# More modern games do exactly this, but they have it as part of the application. Here
+# We add it as a part of the packaging. It was this, or writing a lot of upstream C code.
+- name: ezquake-data
+  buildsystem: simple
+  build-commands:
+  - mkdir /app/share  
+  # required for all pre-packaged files
+  - mv ezquake /app/share/ezquake 
+  - ln -s ~/.ezquake/ezquake /app/bin/ezquake
```
Comment (barthalion) [inline io.github.ezQuake.yaml]: Does that actually work? `build-commands` happen on build time, so `~` will point to different place depending on user used to run `flatpak-builder`.

Comment (barthalion) [reply io.github.ezQuake.yaml]: You actually hit the problem caused by this at https://github.com/flathub/flathub/issues/1175.

I haven't checked myself, but does the binary check it's absolute location or just current working directory?

**Comment by Eonfge** [comment]
bot, build io.github.ezQuake

File: io.github.ezQuake.yaml (hunk 2 above)
Comment (Eonfge) [reply io.github.ezQuake.yaml]: I've changed it using a git-patch. The symlink thing indeed didn't work, but that's one of those things you can't discover when you only compile and run it locally.

On the topic of meta data, I created an issue: https://github.com/ezQuake/ezquake-source/issues/337

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/io.github.ezQuake


Thanks!

**Pull request closed without merging**